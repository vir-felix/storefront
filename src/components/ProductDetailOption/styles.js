export default (theme) => ({
  variantOptionContainer: {
    display: "flex",
    width: `${theme.spacing.unit * 3.125}%`,
    maxWidth: `${theme.spacing.unit * 3.125}%`,
    padding: theme.spacing.unit * 1.25
  },
  optionButton: {
    "flex": "1 1 auto",
    "fontWeight": 600,
    "padding": theme.spacing.unit * 1.25,
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.borderColor,
    "&:hover": {
      backgroundColor: theme.palette.reaction.activeElementBackground
    }
  },
  optionText: {
    fontWeight: 500,
    fontSize: "14px"
  },
  isSelected: {
    backgroundColor: theme.palette.reaction.activeElementBackground
  }
});
