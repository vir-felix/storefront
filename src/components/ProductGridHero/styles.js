export default (theme) => ({
  heroImg: {
    width: "100%",
    height: "325px",
    objectFit: "cover"
  },
  productGridContainer: {
    maxWidth: theme.grid.productGridMaxWidth,
    marginLeft: "auto",
    marginRight: "auto"
  }
});
