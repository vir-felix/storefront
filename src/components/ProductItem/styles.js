export const styles = (theme) => ({
  "@keyframes spin": {
    from: { transform: "rotate(0deg)" },
    to: { transform: "rotate(-360deg)" }
  },
  "root": {},
  "productInfo": {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    paddingTop: 10
  },
  "productMedia": {
    backgroundColor: theme.palette.primary.contrastText,
    position: "relative"
  },
  "img": {
    height: "auto",
    width: "100%"
  },
  "imgLoading": {
    alignItems: "center",
    backgroundColor: "transparent",
    display: "flex",
    fontSize: 48,
    height: 100,
    justifyContent: "center",
    left: "calc(50% - 50px)",
    position: "absolute",
    top: "calc(50% - 50px)",
    width: 100
  },
  "loadingIcon": {
    color: theme.palette.primary.light,
    fontSize: "inherit",
    animationName: "spin",
    animationDuration: theme.transitions.duration.standard * 2,
    animationTimingFunction: theme.transitions.easing.sharp,
    animationIterationCount: "infinite",
    animationFillMode: "both"
  },
  "link": {
    ...theme.typography.body2,
    "textAlign": "left",
    "&:hover": {
      color: theme.palette.secondary.main
    }
  }
});
