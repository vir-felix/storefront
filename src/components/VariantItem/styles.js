export default (theme) => ({
  variantButton: {
    "display": "flex",
    "justifyContent": "space-between",
    "textTransform": "none",
    "width": "100%",
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.contrastText,
    "paddingTop": theme.spacing.unit * 1.25,
    "paddingBottom": theme.spacing.unit * 1.25,
    "paddingLeft": theme.spacing.unit * 2.5,
    "paddingRight": theme.spacing.unit * 2.5,
    "&:hover": {
      border: "1px solid",
      borderColor: theme.palette.reaction.activeElementBorderColor,
      backgroundColor: theme.palette.primary.contrastText
    }
  },
  activeVariant: {
    border: "1px solid",
    borderColor: theme.palette.reaction.activeElementBorderColor
  }
});
