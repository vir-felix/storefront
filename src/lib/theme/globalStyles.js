export default
<style global jsx>
  {`
    a {
      color: inherit;
      text-decoration: none;
    }
  `}
</style>;
