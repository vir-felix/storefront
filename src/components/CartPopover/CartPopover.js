import React, { Component } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import { inject, observer } from "mobx-react";
import Grid from "@material-ui/core/Grid";
import Hidden from "@material-ui/core/Hidden";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import CheckoutButtons from "components/CheckoutButtons";


const styles = (theme) => ({
  container: {
    "alignItems": "center",
    "boxShadow": theme.boxShadow.depth2,
    "display": "flex",
    "marginLeft": "auto",
    "marginRight": "auto",
    "maxWidth": "400px",
    "paddingTop": "12px",
    "position": "fixed",
    "right": 0,
    "top": 0,
    "transitionDuration": "400ms",
    "transitionProperty": "transform",
    "transitionTimingFunction": "linear",
    "backgroundColor": theme.palette.reaction.white,
    "&:hover": {
      transform: "translate(0px, 0px)"
    }
  },
  gridContainer: {
    padding: "10px"
  },
  isContainerHidden: {
    transform: "translate(400px, 0px)"
  },
  isContainerVisible: {
    transform: "translate(0px, 0px)"
  },
  containerItem: {
    alignItems: "center",
    display: "flex"
  },
  addedToCartImg: {
    height: "40px",
    marginRight: "10px",
    width: "40px"
  },
  addedToCartItemName: {
    maxWidth: "200px",
    whiteSpace: "nowrap",
    overflow: "hidden",
    textOverflow: "ellipsis",
    fontWeight: theme.typography.fontWeightMedium,
    display: "inline-block",
    lineHeight: "0.8em"
  },
  addedToCartText: {
    color: theme.palette.primary.dark,
    display: "inline",
    fontSize: theme.typography.fontSize * 0.875
  }
});

@withStyles(styles, { withTheme: true })
@inject("uiStore")
@observer
class CartPopover extends Component {
  static propTypes = {
    /**
     * CartItem data
     */
    cartItem: PropTypes.shape({
      /**
       * The cart item ID
       */
      _id: PropTypes.string,
      /**
       * Array of additional attributes of the chosen item.
       */
      attributes: PropTypes.arrayOf(PropTypes.shape({
        /**
           * Attribute label (i.e. "Color").
           */
        label: PropTypes.string,
        /**
           *  Attribute value (i.e. "Red").
           */
        value: PropTypes.string
      })),
      /**
       * Current stock quantity of item
       */
      currentQuantity: PropTypes.number,
      /**
       * Image url of chosen item
       */
      imageUrl: PropTypes.string,
      /**
       * Is the chosen item have a low quantity
       */
      isLowInventoryQuantity: PropTypes.bool,
      /**
       * Price object of chosen item
       */
      price: PropTypes.shape({
        /**
         * Chosen items compare at price
         */
        compareAtPrice: PropTypes.string,
        /**
         * Chosen items display price
         */
        displayPrice: PropTypes.string
      }),
      /**
       * Chosen items slug
       */
      productSlug: PropTypes.string,
      /**
       * Chosen items title
       */
      title: PropTypes.string,
      /**
       * Quantity of chosen item in cart
       */
      quantity: PropTypes.number
    }).isRequired,
    classes: PropTypes.object,
    isVisible: PropTypes.bool,
    theme: PropTypes.object,
    uiStore: PropTypes.shape({
      isCartPopoverOpen: PropTypes.bool
    }).isRequired
  }

  handleOnClick = () => {
    console.log("go to checkout");
  }

  render() {
    // When cartItem is available as a prop, remove the following lines, and uncomment the prop destructuring below.
    // For now, we are using the static object `cartItem`.
    const { classes: { addedToCartImg, addedToCartItemName, addedToCartText, container, containerItem, gridContainer, isContainerHidden, isContainerVisible }, theme, uiStore } = this.props;
    const cartItem = {
      _id: "abcdefghijklmnop",
      attributes: [
        {
          label: "Color",
          value: "Red"
        },
        {
          label: "Season",
          value: "Summer"
        }
      ],
      currentQuantity: 10,
      imageUrl: "http://localhost:3000/assets/files/Media/nbYKLrZST5DSF87md/thumbnail/chuttersnap-265339-unsplash.png",
      isLowInventoryQuantity: false,
      price: {
        compareAtPrice: "19.99",
        displayPrice: "18.99"
      },
      productSlug: "product-slug",
      title: "Item Title",
      quantity: 10
    };
    // Remove the above lines, and uncomment line below when `cartItem` is available as a prop
    // const { cartItem, classes: { addedToCartImg, addedToCartItemName, addedToCartText, container, containerItem, gridContainer, isContainerHidden, isContainerVisible }, theme, uiStore } = this.props;

    return (
      <Hidden mdUp>
        <div className={classNames(container, { [isContainerHidden]: !uiStore.isCartPopoverOpen, [isContainerVisible]: uiStore.isCartPopoverOpen })}>
          <Grid container className={gridContainer} spacing={theme.spacing.unit * 3}>
            <Grid className={containerItem} item xs={12}>
              <img alt={cartItem.title} className={addedToCartImg} src={cartItem.imageUrl} />
              <Typography className={addedToCartText} component="span">
                {cartItem.quantity} "<span className={addedToCartItemName}>{cartItem.title}</span>" added to cart
              </Typography>
            </Grid>
            <Grid item xs={12}>
              <CheckoutButtons
                onClick={this.handleOnClick}
              />
            </Grid>
          </Grid>
        </div>
      </Hidden>
    );
  }
}

export default CartPopover;
