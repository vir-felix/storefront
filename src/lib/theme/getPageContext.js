import { SheetsRegistry } from "jss";
import { createGenerateClassName } from "@material-ui/core/styles";
import theme from "./reactionTheme";

/**
 * @summary Creates and returns a page context object
 * @returns {Object} Page context object
 */
function createPageContext() {
  return {
    theme,
    // This is needed in order to deduplicate the injection of CSS in the page.
    sheetsManager: new Map(),
    // This is needed in order to inject the critical CSS.
    sheetsRegistry: new SheetsRegistry(),
    // The standard class name generator.
    generateClassName: createGenerateClassName()
  };
}

/**
 * Either creates a new or returns an existing page context object. If called from Node, creates
 * a new object every time. Otherwise creates and caches an object in the browser and
 * subsequent calls will return that.
 *
 * @summary Gets a page context object
 * @returns {Object} Page context object
 */
export default function getPageContext() {
  // Make sure to create a new context for every server-side request so that data
  // isn't shared between connections (which would be bad).
  if (!process.browser) {
    return createPageContext();
  }

  // Reuse context on the client-side.
  if (!global.__INIT_MATERIAL_UI__) {
    global.__INIT_MATERIAL_UI__ = createPageContext();
  }

  return global.__INIT_MATERIAL_UI__;
}
