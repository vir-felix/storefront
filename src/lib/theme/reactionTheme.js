import { createMuiTheme } from "material-ui/styles";

const theme = createMuiTheme({
  palette: {
    primary: {
      light: "#26B0F9",
      main: "#1999DD",
      dark: "#172F3C",
      contrastText: "#FFFFFF",
      borderColor: "#CCCCCC"
    },
    secondary: {
      light: "#5d8ea9",
      main: "#5E7480",
      dark: "#1D1D1D",
      contrastText: "#000000"
    },
    error: {
      light: "#E54F5D",
      main: "#CD3F4C",
      dark: "#3C1F21",
      contrastText: "#FFFFFF"
    },
    reaction: {
      activeElementBorderColor: "#94E8D1",
      activeElementBackground: "#E6E6E6"
    }

  },
  typography: {
    fontFamily: "Source Sans Pro, Helvetica Neue, Helvetica, sans-serif",
    fontSize: 16,
    fontWeightLight: 200,
    fontWeightRegular: 400,
    fontWeightMedium: 700,
    fontWeightBold: 700
  }
});

export default theme;
