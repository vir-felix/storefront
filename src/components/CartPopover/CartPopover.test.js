import React from "react";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "@material-ui/core/styles";
import { Provider } from "mobx-react";
import theme from "lib/theme/reactionTheme";
import CartPopover from "./CartPopover";

const cartItem = {
  _id: "abcdefghijklmnop",
  attributes: [
    {
      label: "Color",
      value: "Red"
    },
    {
      label: "Season",
      value: "Summer"
    }
  ],
  currentQuantity: 10,
  imageUrl: "http://localhost:3000/assets/files/Media/nbYKLrZST5DSF87md/thumbnail/chuttersnap-265339-unsplash.png",
  isLowInventoryQuantity: false,
  price: {
    compareAtPrice: "19.99",
    displayPrice: "18.99"
  },
  productSlug: "product-slug",
  title: "Item Title",
  quantity: 10
};

const uiStore = {
  isCartPopoverOpen: true
};

const uiStoreFalse = {
  isCartPopoverOpen: false
};

test("show cartpopover", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <CartPopover cartItem={cartItem} uiStore={uiStore} />
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("do not show cartpopover", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <CartPopover cartItem={cartItem} uiStore={uiStoreFalse} />
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

