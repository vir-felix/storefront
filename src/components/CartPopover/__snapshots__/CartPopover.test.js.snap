// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`do not show cartpopover 1`] = `
<div
  className="HiddenCss-mdUp-16"
>
  <div
    className="inject-CartPopover-with-uiStore-container-1 inject-CartPopover-with-uiStore-isContainerHidden-3"
  >
    <div
      className="MuiGrid-container-24 MuiGrid-spacing-xs-24-47 inject-CartPopover-with-uiStore-gridContainer-2"
    >
      <div
        className="MuiGrid-item-25 MuiGrid-grid-xs-12-62 inject-CartPopover-with-uiStore-containerItem-5"
      >
        <img
          alt="Item Title"
          className="inject-CartPopover-with-uiStore-addedToCartImg-6"
          src="//placehold.it/100"
        />
        <span
          className="MuiTypography-root-115 MuiTypography-body1-124 inject-CartPopover-with-uiStore-addedToCartText-8"
        >
          10
           "
          <span
            className="inject-CartPopover-with-uiStore-addedToCartItemName-7"
          >
            Item Title
          </span>
          " added to cart
        </span>
      </div>
      <div
        className="MuiGrid-item-25 MuiGrid-grid-xs-12-62"
      >
        <a
          className="WithTracking-Link--link-139"
          href="/checkout"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="link"
          tabIndex={0}
          tracking={
            Object {
              "getTrackingData": [Function],
              "trackEvent": [Function],
            }
          }
        >
          <div
            className="Button__ButtonDiv-bnfrK eloVkm"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
            title={undefined}
          >
            <div>
              Checkout
            </div>
            <div
              className="Button__SpinnerWrap-bgOcuM cCKgPo"
              style={
                Object {
                  "opacity": 0,
                  "paddingLeft": 0,
                  "width": 0,
                }
              }
            >
              <svg
                className="spinner__SpinnerSvg-iBTUvT bunzNs"
                height="40px"
                style={
                  Object {
                    "enableBackground": "new 0 0 50 50",
                  }
                }
                version="1.1"
                viewBox="0 0 50 50"
                width="40px"
                x="0px"
                y="0px"
              >
                <path
                  d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
                  fill="#000"
                >
                  <animateTransform
                    attributeName="transform"
                    attributeType="xml"
                    dur="1s"
                    from="0 25 25"
                    repeatCount="indefinite"
                    to="360 25 25"
                    type="rotate"
                  />
                </path>
              </svg>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`show cartpopover 1`] = `
<div
  className="HiddenCss-mdUp-16"
>
  <div
    className="inject-CartPopover-with-uiStore-container-1 inject-CartPopover-with-uiStore-isContainerVisible-4"
  >
    <div
      className="MuiGrid-container-24 MuiGrid-spacing-xs-24-47 inject-CartPopover-with-uiStore-gridContainer-2"
    >
      <div
        className="MuiGrid-item-25 MuiGrid-grid-xs-12-62 inject-CartPopover-with-uiStore-containerItem-5"
      >
        <img
          alt="Item Title"
          className="inject-CartPopover-with-uiStore-addedToCartImg-6"
          src="//placehold.it/100"
        />
        <span
          className="MuiTypography-root-115 MuiTypography-body1-124 inject-CartPopover-with-uiStore-addedToCartText-8"
        >
          10
           "
          <span
            className="inject-CartPopover-with-uiStore-addedToCartItemName-7"
          >
            Item Title
          </span>
          " added to cart
        </span>
      </div>
      <div
        className="MuiGrid-item-25 MuiGrid-grid-xs-12-62"
      >
        <a
          className="WithTracking-Link--link-139"
          href="/checkout"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="link"
          tabIndex={0}
          tracking={
            Object {
              "getTrackingData": [Function],
              "trackEvent": [Function],
            }
          }
        >
          <div
            className="Button__ButtonDiv-bnfrK eloVkm"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
            title={undefined}
          >
            <div>
              Checkout
            </div>
            <div
              className="Button__SpinnerWrap-bgOcuM cCKgPo"
              style={
                Object {
                  "opacity": 0,
                  "paddingLeft": 0,
                  "width": 0,
                }
              }
            >
              <svg
                className="spinner__SpinnerSvg-iBTUvT bunzNs"
                height="40px"
                style={
                  Object {
                    "enableBackground": "new 0 0 50 50",
                  }
                }
                version="1.1"
                viewBox="0 0 50 50"
                width="40px"
                x="0px"
                y="0px"
              >
                <path
                  d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
                  fill="#000"
                >
                  <animateTransform
                    attributeName="transform"
                    attributeType="xml"
                    dur="1s"
                    from="0 25 25"
                    repeatCount="indefinite"
                    to="360 25 25"
                    type="rotate"
                  />
                </path>
              </svg>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
